workflows/cloudify_system_workflows/snapshot.py

```
@@ -0,0 +1,281 @@
+                     '?u=root&p=root&chunked=true" --data-urlencode'
+                     ' "q=select * from /.*/" > {0}')
+INFLUXDB_RESTORE_CMD = ('cat {0} | while read -r line; do curl -X POST '
+                        '-d "[${{line}}]" "http://localhost:8086/db/cloudify/'
+                        'series?u=root&p=root" ;done')
+
+
+class DictToAttributes(object):
+    def __init__(self, dic):
+        self._dict = dic
+
+    def __getattr__(self, name):
+        return self._dict[name]
+
+
+def get_json_objects(f):
+    def chunks(g):
+        ch = g.read(10000)
+        yield ch
+        while ch:
+            ch = g.read(10000)
+            yield ch
+
+    s = ''
+    decoder = json.JSONDecoder()
+    for ch in chunks(f):
+        s += ch
+        try:
+            while s:
+                obj, idx = decoder.raw_decode(s)
+                yield json.dumps(obj)
+                s = s[idx:]
+        except:
+            pass
+
+    assert not s
+
+
+def copy_data(archive_root, config, to_archive=True):
+    DATA_TO_COPY = [
+        (config.file_server_blueprints_folder, 'blueprints'),
+        (config.file_server_uploaded_blueprints_folder, 'uploaded-blueprints')
+    ]
+
+    # files with constant relative/absolute paths
+    for (p1, p2) in DATA_TO_COPY:
+        if p1[0] != '/':
+            p1 = path.join(config.file_server_root, p1)
+        if p2[0] != '/':
+            p2 = path.join(archive_root, p2)
+        if not to_archive:
+            p1, p2 = p2, p1
+
+        if not path.exists(p1):
+            continue
+
+        if path.isfile(p1):
+            shutil.copy(p1, p2)
+        else:
+            if path.exists(p2):
+                shutil.rmtree(p2, ignore_errors=True)
+            shutil.copytree(p1, p2)
+
+
+def _create_es_client(config):
+    return elasticsearch.Elasticsearch(hosts=[{'host': config.db_address,
+                                               'port': config.db_port}])
+
+
+def _except_types(s, *args):
```

for now, let's raise an error if the new manager is not empty